### PR TITLE
chore: update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by dgozman**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,92 +1,77 @@
...
-npm i -g npm@latest
 ```
 
-1. Clone this repository
-
-    ```bash
-    git clone https://github.com/microsoft/playwright
-    cd playwright
-    ```
+Clone the repository. If you plan to send a pull request, it might be better to [fork the repository](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/fork-a-repo) first.
```
Comment (mxschmitt) [inline CONTRIBUTING.md:16]: Usually you clone our repository and then add your fork as an origin / VSCode will automatically fork the repository for you if try to push to microsoft. I'd not mention forking here at all.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -97,131 +82,57 @@ footer
...
 ```
-fix(firefox): make sure session cookies work
+feat(trace viewer): network panel filtering
 
-This patch fixes session cookies in the firefox browser.
+This patch adds a filtering toolbar to the network panel.
+<link to a screenshot>
 
-Fixes #123, fixes #234
+Fixes #123, references #234.
```
Comment (mxschmitt) [inline CONTRIBUTING.md:104]:
```suggestion
Fixes #123
References #234
```

**Review by mxschmitt — APPROVED** [review]

**Comment by dgozman** [comment]
@pengooseDev Thank you!

> All the numbering is set to 1, just wanted to point that out in case it's unintentional

That's intentional, markdown handles it just right.

> I initially had difficulty since it's my first time seeing dynamically generated docs (e.g., docs/src/test-api) and types (e.g., utils/generate_types).

There is the following line currently, but it's not very prominent:
`Note that some files are generated by the build, so the watch process might override your changes if done in the wrong file. For example, TypeScript types for the API are generated from the [docs/src](https://github.com/microsoft/playwright/blob/main/docs/src)`

If you have any specific ideas on improving the guide, I would love to review a PR!

**Comment by pengooseDev** [comment]
@dgozman 
The `CONTRIBUTING` guide you've written already does a great job of outlining the process of creating a PR by effectively using header(`##`). 😀
 I believe adding subHeader(###) under the `## Make a change` section could further emphasize key points and make the flow of code changes more explicit.
 
 This is just a tiny suggestion, so feel free to disregard it if it doesn't align with teams visions. :) Thx!

## Example

```md
### TypeScript Type Generation

If your changes involve TypeScript types, ensure to modify the corresponding files in [`docs/src`](https://github.com/microsoft/playwright/blob/main/docs/src).
Note that some files are generated by the build, so the watch process might override your changes if done in the wrong file. For example, TypeScript types for the API are generated from this directory.

### Linting

Coding style is fully defined in [.eslintrc](https://github.com/microsoft/playwright/blob/main/.eslintrc.js). Before creating a pull request, or at any moment during development, run linter to check all kinds of things:
  ```bash
  npm run lint
  ```
```

**Pull request merged**